instructure/canvas-lms
# replace slashes with underscores in gradebook export filename

When using local storage instead of S3, having a course short name with a slash in it will cause an error (No such file or directory @ rb_sysopen) when trying to export the Gradebook as a CSV. attachment_fu is receiving the slash as part of the file path, and erroring out because it can't find the file.

A CLA is on file under my employer, Simon Fraser University (github.com/sfu). Happy to review this in person at Instructurecon :)

Test plan:

  - enable local storage for canvas
  - create a course with a slash in the short name
    (e.g. CHEM 121 / CHEM 122)
  - attempt to export the gradebook
  - ensure the gradebook file was exported and downlaoded correctly
